src/App.tsx

```diff
@@ -1,24 +1,268 @@
+    event.preventDefault();
+
+    if (inputValue.trim() !== '') {
+      const tempTodoItem: Todo = {
+        title: inputValue.trim(),
+        userId: USER_ID,
+        completed: false,
+        id: getRandomNumber(),
+      };
+
+      try {
+        await client.post('/todos', tempTodoItem);
+        setTodo((prevTodo) => [...prevTodo, tempTodoItem]);
+      } catch (error) {
+        console.log('There is an error', error);
+      }
+    } else {
+      // setErrorMessageField(true);
+    }
+
+    setInputValue('');
+  };
+
+  const deleteTodo = async (id: number) => {
+    const tempTodos = todo.filter((element) => {
+      return element.id !== id;
+    });
+
+    setTodo(tempTodos);
+
+    try {
+      await client.delete(`/todos/${id}`);
+    } catch (error) {
+      console.log('There is an issue.', error);
+      setDeleteErrorMessage('Unable to delete a todo');
+    }
+  };
+
+  const updateAllTodo = async () => {
+    const everyCompleted = todo.every((item) => item.completed);
+
+    const updatedTodos = todo.map((element) => {
+      return everyCompleted
+        ? ({
+          ...element,
+          completed: !element.completed,
+        }) : (
+          {
+            ...element,
+            completed: true,
+          }
+        );
+    });
+
+    setTodo(updatedTodos);
+
+    try {
+      updatedTodos.map(async (todoItem) => {
+        await client.patch(`/todos/${todoItem.id}`, {
+          ...todoItem,
+          completed: todoItem.completed,
+        });
+      });
+    } catch (error) {
+      console.log('There is an issue updating todos.', error);
+    }
+  };
+
+  const updateIndividualTodo = async (id: number) => {
+    const updatedTodo = todo.map((obj) => {
+      if (obj.id === id) {
+        return {
+          ...obj,
+          completed: !obj.completed,
+        };
+      }
+
+      return obj;
+    });
+
+    const none = todo.some((element) => {
+      return element.id === id;
+    });
+
+    if (!none) {
+      // setIsEditingTodoAllowed(true);
+      // setErrorMessageField(true);
+      setEditTodo('Unable to loading a todo');
+      setIsThereIssue(true);
+    }
+
+    setTodo(updatedTodo);
+
+    try {
+      const todoToUpdate = todo.find((elem) => elem.id === id);
+
+      if (todoToUpdate) {
+        await client.patch(`/todos/${id}`, {
+          completed: !todoToUpdate.completed,
+          title: todoToUpdate.title,
+          userId: USER_ID,
+          id,
+        });
+      }
+    } catch (error) {
+      setDeleteErrorMessage('Unable to delete a todo');
+      console.log('There is an issue.');
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+
+    return () => {
+      if (timeoutId.current) {
+        clearTimeout(timeoutId.current);
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    const isActive = todo.some((obj) => obj.completed === false);
+    const isCompleted = todo.some((obj) => obj.completed === true);
```

Comment:
Use logical operators, please 
```suggestion
    const isActive = todo.some((obj) => !obj.completed);
    const isCompleted = todo.some((obj) => obj.completed);
```